sql/memo/select_hints_test.go

```diff
@@ -172,9 +171,14 @@ func TestOrderHintBuilding(t *testing.T) {
 				10: testVertexSet(0, 1, 2),    // ab x xy x pq
 				11: testVertexSet(3),          // uv
 				14: testVertexSet(0, 1, 2, 3), // ab x xy x pq x uv
-				15: testVertexSet(1, 2),       // xy x pq
-				16: testVertexSet(2, 3),       // pq x uv
-				17: testVertexSet(1, 2, 3),    // xy x pq x uv
+				18: testVertexSet(0, 2),       // ab x pq
```

Comment:
On the same theme, these would be more useful if the tests were defined in terms of the comments and you did translation via some function

As these stand, they're brittle: they're going to break when we fix a bug or make other changes, and it's going to be very hard for someone to figure out why

For example it's super non-obvious why the group ids in these expected values aren't contiguous. If the answer is "that's a detail of the way we build the data structure" fine, but that should be abstracted away from how we build the test expectation data in that case. This kind of definition leads to people just regenerating the tests because they don't understand how to construct them.

Reply:
yeah i'll rewrite, i'm going to end up reversing this and making these contiguous again soon i think